components/do_verify_email/do_verify_email.jsx

```diff
@@ -25,30 +31,29 @@ export default class DoVerifyEmail extends React.Component {
     }
 
     componentWillMount() {
-        verifyEmail(
-            this.props.location.query.token,
-            () => {
-                browserHistory.push('/login?extra=verified&email=' + encodeURIComponent(this.props.location.query.email));
-            },
-            (err) => {
-                let serverError;
-                if (err) {
-                    serverError = (
-                        <FormattedMessage
-                            id='signup_user_completed.invalid_invite'
-                            defaultMessage='The invite link was invalid.  Please speak with your Administrator to receive an invitation.'
-                        />
-                    );
-                }
-
-                this.setState({
-                    verifyStatus: 'failure',
-                    serverError
-                });
-            }
-        );
+        this.verifyEmail();
     }
 
+    verifyEmail = async () => {
+        const {actions: {verifyUserEmail}} = this.props;
+        const {data, error} = await verifyUserEmail(this.props.location.query.token);
```

Comment:
Don't know why but this is not playing nicely with the test. It works fine though with below code but I'm 0/5 if it's a good pattern:
```
        const verify = await verifyUserEmail(this.props.location.query.token);

       if (!verify) {
            // something wrong happened
        }else if (verify.data) {
            ...
        } else if (verify.error) {
            ...
        }
```

Reply:
Everything else you mentioned worked really well. Thank you for the help.

but regarding this part, I tried to see if maybe there was a better pattern to use, and the best I could come up with ended up being: 
```
const verify = await verifyUserEmail(this.props.location.query.token);

        if (verify && verify.data) {
            browserHistory.push('/login?extra=verified&email=' + encodeURIComponent(this.props.location.query.email));
        } else if (verify && verify.error) {
            const serverError = (
                <FormattedMessage
                    id='signup_user_completed.invalid_invite'
                    defaultMessage='The invite link was invalid.  Please speak with your Administrator to receive an invitation.'
                />
            );
            this.setState({
                verifyStatus: 'failure',
                serverError
            });
        }
```
Which isn't too different than what you showed.

I'll keep trying to find a possibly better pattern for this, let me know what you think

Reply:
Cool, I'm good with it.